src/Microsoft.DotNet.XHarness.WebAssembly/xunit-runner.cs

```diff
@@ -0,0 +1,354 @@
+
+            if (tc is XunitTheoryTestCase)
+            {
+                if (method.IsGenericMethod)
+                {
+                    Console.WriteLine("SKIP (generic): " + tc.DisplayName);
+                    nfiltered++;
+                    return;
+                }
+
+                // From XunitTheoryTestCaseRunner
+                IEnumerable<IAttributeInfo> attrs = tc.TestMethod.Method.GetCustomAttributes(typeof(DataAttribute));
+                bool failed = false;
+                foreach (IAttributeInfo dataAttribute in attrs)
+                {
+                    IAttributeInfo discovererAttribute = dataAttribute.GetCustomAttributes(typeof(DataDiscovererAttribute)).First();
+                    var args = discovererAttribute.GetConstructorArguments().Cast<string>().ToList();
+
+                    Type discovererType = null;
+                    if (args[1] == "xunit.core")
+                        discovererType = typeof(IXunitTestCollectionFactory).Assembly.GetType(args[0]);
+                    if (discovererType == null)
+                    {
+                        Console.WriteLine("FAIL (discoverer): " + args[0] + " " + args[1]);
+                        failed = true;
+                    }
+
+                    IDataDiscoverer discoverer;
+                    discoverer = ExtensibilityPointFactory.GetDataDiscoverer(this, discovererType);
+
+                    try
+                    {
+                        IEnumerable<object[]> data = discoverer.GetData(dataAttribute, tc.TestMethod.Method);
+                        object[][] data_arr = data.ToArray();
+                        Console.WriteLine(tc.DisplayName + " [" + data_arr.Length + "]");
+                        foreach (object[] dataRow in data_arr)
+                        {
+                            nrun++;
+                            object obj = null;
+                            if (!method.IsStatic)
+                            {
+                                ConstructorInfo constructor = method.ReflectedType.GetConstructor(Type.EmptyTypes);
+                                if (constructor != null)
+                                    obj = constructor.Invoke(null);
+                                else
+                                    obj = System.Runtime.Serialization.FormatterServices.GetUninitializedObject(method.ReflectedType);
+                            }
+
+                            ParameterInfo[] pars = method.GetParameters();
+                            for (int i = 0; i < dataRow.Length; ++i)
+                                dataRow[i] = ConvertArg(dataRow[i], pars[i].ParameterType);
+                            method.Invoke(obj, BindingFlags.Default, null, dataRow, null);
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Console.WriteLine("FAIL: " + tc.DisplayName + " " + ex);
+                        failed = true;
```

Comment:
Do we want to return here?

Reply:
I guess we cannot because of the counter below?

Reply:
yeah, this should the reason